src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -57,22 +57,22 @@ public class BatchExportProcessor<T> : BaseExportProcessor<T>
         {
             if (maxQueueSize <= 0)
             {
-                throw new ArgumentOutOfRangeException(nameof(maxQueueSize));
+                throw new ArgumentOutOfRangeException(nameof(maxQueueSize), maxQueueSize, $"Max queue size should be greater than 0, but was: {maxQueueSize}");
             }
 
             if (maxExportBatchSize <= 0 || maxExportBatchSize > maxQueueSize)
             {
-                throw new ArgumentOutOfRangeException(nameof(maxExportBatchSize));
+                throw new ArgumentOutOfRangeException(nameof(maxExportBatchSize), maxExportBatchSize, $"Max export batch size should be above 0 or below max queue size. Max queue size was: {maxQueueSize} and Max export batch size was: {maxExportBatchSize}");
```

Comment:
```suggestion
                throw new ArgumentOutOfRangeException(nameof(maxExportBatchSize), maxExportBatchSize, $"maxExportBatchSize should be above 0 and below maxQueueSize. maxQueueSize was: {maxQueueSize} and maxExportBatchSize was: {maxExportBatchSize}");
```

Reply:
I suggest that we put the actual parameter name as it is easier to do the mental mapping

Reply:
An alternative way is: `maxExportBatchSize must be in range (0, maxQueueSize].`

Reply:
Fixed this in latests commit.